### ISSUE TITLE
[WSL2 Ubuntu] rust-analyzer vscode extension suddenly fails to start

### ISSUE THREAD

**Issue by dacite** [rust-lang/rust-analyzer#12367]
Starting today, rust-analyzer (v0.4.1067) failed to start due to the following error:
```
Unfortunately we don't ship binaries for your platform yet. You need to manually clone the rust-analyzer repository and run `cargo xtask install --server` to build the language server from sources. If you feel that your platform should be supported, please create an issue about that [here](https://github.com/rust-analyzer/rust-analyzer/issues) and we will consider it.

```
This message seems misleading because I've been running nightly version with no issues until today.
Although running `require('os').platform()` inside vscode returns `win32`.

<details>
  <summary>Client Log</summary>
<code>
INFO [5/23/2022, 2:56:40 PM]: Extension version: 0.4.1067
INFO [5/23/2022, 2:56:40 PM]: Using configuration {
  cargoRunner: null,
  runnableEnv: null,
  server: { path: null, extraEnv: null },
  trace: { server: 'off', extension: true },
  debug: {
    engine: 'auto',
    sourceFileMap: {
      '/rustc/<id>': '${env:USERPROFILE}/.rustup/toolchains/<toolchain-id>/lib/rustlib/src/rust'
    },
    openDebugPane: false,
    engineSettings: {}
  },
  assist: { expressionFillDefault: 'todo' },
  cachePriming: { enable: true, numThreads: 0 },
  cargo: {
    autoreload: true,
    buildScripts: { enable: true, overrideCommand: null, useRustcWrapper: true },
    features: [],
    noDefaultFeatures: false,
    noSysroot: false,
    target: null,
    unsetTest: [ 'core' ],
    allFeatures: true
  },
  checkOnSave: {
    allTargets: true,
    command: 'check',
    enable: true,
    extraArgs: [],
    features: null,
    noDefaultFeatures: null,
    overrideCommand: null,
    target: null
  },
  completion: {
    autoimport: { enable: true },
    autoself: { enable: true },
    callable: { snippets: 'fill_arguments' },
    postfix: { enable: true },
    privateEditable: { enable: false },
    snippets: {
      custom: {
        'Arc::new': {
          postfix: 'arc',
          body: 'Arc::new(${receiver})',
          requires: 'std::sync::Arc',
          description: 'Put the expression into an `Arc`',
          scope: 'expr'
        },
        'Rc::new': {
          postfix: 'rc',
          body: 'Rc::new(${receiver})',
          requires: 'std::rc::Rc',
          description: 'Put the expression into an `Rc`',
          scope: 'expr'
        },
        'Box::pin': {
          postfix: 'pinbox',
          body: 'Box::pin(${receiver})',
          requires: 'std::boxed::Box',
          description: 'Put the expression into a pinned `Box`',
          scope: 'expr'
        },
        Ok: {
          postfix: 'ok',
          body: 'Ok(${receiver})',
          description: 'Wrap the expression in a `Result::Ok`',
          scope: 'expr'
        },
        Err: {
          postfix: 'err',
          body: 'Err(${receiver})',
          description: 'Wrap the expression in a `Result::Err`',
          scope: 'expr'
        },
        Some: {
          postfix: 'some',
          body: 'Some(${receiver})',
          description: 'Wrap the expression in an `Option::Some`',
          scope: 'expr'
        }
      }
    }
  },
  diagnostics: {
    disabled: [],
    enable: true,
    experimental: { enable: false },
    remapPrefix: {},
    warningsAsHint: [],
    warningsAsInfo: []
  },
  files: { excludeDirs: [], watcher: 'client' },
  highlightRelated: {
    breakPoints: { enable: true },
    exitPoints: { enable: true },
    references: { enable: true },
    yieldPoints: { enable: true }
  },
  hover: {
    actions: {
      debug: { enable: true },
      enable: true,
      gotoTypeDef: { enable: true },
      implementations: { enable: true },
      references: { enable: false },
      run: { enable: true }
    },
    documentation: { enable: true },
    links: { enable: true }
  },
  imports: {
    granularity: { enforce: false, group: 'crate' },
    group: { enable: true },
    merge: { glob: true },
    prefix: 'plain'
  },
  inlayHints: {
    bindingModeHints: { enable: false },
    chainingHints: { enable: true },
    closingBraceHints: { enable: true, minLines: 25 },
    closureReturnTypeHints: { enable: false },
    lifetimeElisionHints: { enable: 'never', useParameterNames: false },
    maxLength: 25,
    parameterHints: { enable: true },
    reborrowHints: { enable: 'never' },
    renderColons: true,
    typeHints: {
      enable: true,
      hideClosureInitialization: false,
      hideNamedConstructor: false
    },
    enabled: false,
    enable: false
  },
  joinLines: {
    joinAssignments: true,
    joinElseIf: true,
    removeTrailingComma: true,
    unwrapTrivialBlock: true
  },
  lens: {
    debug: { enable: true },
    enable: true,
    forceCustomCommands: true,
    implementations: { enable: true },
    references: {
      adt: { enable: false },
      enumVariant: { enable: false },
      method: { enable: false },
      trait: { enable: false }
    },
    run: { enable: true }
  },
  linkedProjects: [],
  lru: { capacity: null },
  notifications: { cargoTomlNotFound: true },
  procMacro: {
    attributes: { enable: true },
    enable: true,
    ignored: {},
    server: null
  },
  runnables: { command: null, extraArgs: [] },
  rustc: { source: null },
  rustfmt: {
    extraArgs: [],
    overrideCommand: null,
    rangeFormatting: { enable: false }
  },
  semanticHighlighting: { strings: { enable: true } },
  signatureInfo: { detail: 'full', documentation: { enable: true } },
  workspace: {
    symbol: { search: { kind: 'only_types', limit: 128, scope: 'workspace' } }
  }
}
</code>
</details>

This issue can be fixed by running the release version. Anyway to fix this issue for nightly?

Not sure if this is helpful but checking /home/dacite/.vscode-server/extensions/ revealed that the platform info is truncated: 
![image](https://user-images.githubusercontent.com/98002119/169913217-d7bd0600-b8cd-480c-ac95-00b5a25dcbfb.png)

Manually renaming it to have the platform name did not fix it. A similar issue is also mentioned [here](https://github.com/rust-lang/rust-analyzer/issues/12356#issuecomment-1135024081)

**Comment by Veykril** [comment]
We published an extension version with no platform binaries to fix another issue with VSCode. This should fix itself with the next nightly hopefully (in ~2hrs). Meanwhile as you've said you can either install the release version or specifically install the previous nightly version from VSCode. Both of those should work.

**Comment by lnicola** [comment]
> We published an extension version with no platform binaries to fix another issue with VSCode.

Well, we didn't actually intend to publish it without the platform VSIXes, the upload just failed because of another issue 😅.
